tests/run-make/comment-section/rmake.rs

```diff
@@ -0,0 +1,41 @@
+// Both GCC and Clang write by default a `.comment` section with compiler information.
+// Rustc received a similar .comment section, so this tests checks that this section
+// properly appears.
+// See https://github.com/rust-lang/rust/commit/74b8d324eb77a8f337b35dc68ac91b0c2c06debc
+
+//@ only-linux
+
+use std::path::PathBuf;
+
+use run_make_support::llvm_readobj;
+use run_make_support::rustc;
+use run_make_support::{cwd, env_var, read_dir, run_in_tmpdir};
+
+fn main() {
+    let target = env_var("TARGET");
+    let p = cwd();
+
+    rustc().input("main.rs").emit("link,obj").arg("-Csave-temps").target(&target).run();
+
+    // Check linked output has a `.comment` section with the expected content.
+    llvm_readobj()
+        .gnu_elf_style()
+        .section(".comment")
+        .arg(PathBuf::from(&p).join("main"))
```

Comment:
Suggestions:

- I believe we can just use `"main"` here because the `main` binary is produced and placed in the `cwd()` as well (where the recipe is executed).
- We can add a more semantically meaningful `input<P: AsRef<Path>>(path: P)` helper which just forwards to `arg(path.as_ref())`.

```suggestion
        .input("main")
```